src/NServiceBus.Core/Pipeline/Incoming/MessageProcessingAbortedException.cs

```diff
@@ -1,8 +1,30 @@
 namespace NServiceBus
 {
     using System;
+    using System.Runtime.Serialization;
+    using JetBrains.Annotations;
+    using NServiceBus.Features;
+    using NServiceBus.Transports;
 
-    class MessageProcessingAbortedException : Exception
+    /// <summary>
+    /// This exception can be thrown by any of recoverability behaviors ie.e <see cref="FirstLevelRetries"/>, <see cref="SecondLevelRetries"/> or <see cref="StoreFaultsInErrorQueue"/>
```

Comment:
Still don't see the value of mentioning the behaviors or even recoverability at all. The doco is there to tell the transport implementor what he should do?

I.e. roll the message back to the queue and do not trigger the circuit breaker?

This made me realize that we should add this to the `NServiceBus.Transports` namespace